server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocatorTests.java

```diff
@@ -235,6 +235,78 @@ public void testRebalanceImprovesTheBalanceOfTheShards() {
         }
     }
 
+    public void testGetIndexDiskUsageInBytes() {
+        {
+            final var indexDiskUsageInBytes = getIndexDiskUsageInBytes(
+                ClusterInfo.EMPTY,
+                IndexMetadata.builder("index").settings(settings(Version.CURRENT)).numberOfShards(1).numberOfReplicas(0).build()
+            );
+
+            assertThat(indexDiskUsageInBytes, is(equalTo(0L)));
+        }
+
+        {
+            final var shardSize = ByteSizeValue.ofGb(50).getBytes();
+
+            final Map<String, Long> shardSizes = new HashMap<>();
+            shardSizes.put("[index][0][p]", shardSize);
+            shardSizes.put("[index][0][r]", shardSize - randomLongBetween(0, 10240));
+
+            final var indexDiskUsageInBytes = getIndexDiskUsageInBytes(
+                randomBoolean() ? ClusterInfo.EMPTY : new ClusterInfo(Map.of(), Map.of(), shardSizes, Map.of(), Map.of(), Map.of()),
+                IndexMetadata.builder("heavy-index")
+                    .settings(settings(Version.CURRENT))
+                    .numberOfShards(1)
+                    .numberOfReplicas(1)
+                    .shardSizeInBytesForecast(shardSize)
+                    .build()
+            );
+
+            assertThat(indexDiskUsageInBytes, is(equalTo(shardSize * 2)));
+        }
+
+        {
+            final var shardSize = ByteSizeValue.ofGb(50).getBytes();
+
+            final Map<String, Long> shardSizes = new HashMap<>();
+            shardSizes.put("[index][0][p]", shardSize);
+            shardSizes.put("[index][0][r]", shardSize - randomLongBetween(0, 10240));
+
+            final var indexDiskUsageInBytes = getIndexDiskUsageInBytes(
+                new ClusterInfo(Map.of(), Map.of(), shardSizes, Map.of(), Map.of(), Map.of()),
+                IndexMetadata.builder("index").settings(settings(Version.CURRENT)).numberOfShards(1).numberOfReplicas(1).build()
+            );
+
+            assertThat(indexDiskUsageInBytes, is(equalTo(shardSizes.values().stream().mapToLong(size -> size).sum())));
+        }
+
+        {
+            // Only 2 of 4 shards sizes are available, therefore an average is calculated
+            // in order to compute the total index size
```

Comment:
It would be great to add similar short description to other scenarios as well or may be even split into a separate test methods (I wish we have junit 5 to still have them grouped).